test/simulator/misc/decisions.js

```diff
@@ -726,15 +726,17 @@ describe('Choice extensions', function () {
 	describe('Undo', function () {
 		const MODES = ['revoke', 'override'];
 		for (const mode of MODES) {
-			it(`should disallow to ${mode} decisions after every player has sent an unrevoked action`, function () {
+			// FIXME
+			it.skip(`should disallow to ${mode} decisions after every player has sent an unrevoked action`, function () {
 				battle = common.createBattle({cancel: true});
 				battle.setPlayer('p1', {team: [{species: "Bulbasaur", ability: 'overgrow', moves: ['tackle', 'growl']}]});
 				battle.setPlayer('p2', {team: [{species: "Charmander", ability: 'blaze', moves: ['tackle', 'growl']}]});
 
 				battle.choose('p1', 'move tackle');
 				battle.choose('p2', 'move growl');
-				if (mode === 'revoke') battle.undoChoice('p1');
-				battle.choose('p1', 'move growl');
+				assert(battle.p1.lastRequest.noCancel);
```

Comment:
The next turn has already started, so ``noCancel`` is falsey, and ``battle.undoChoice()`` is noop.

Reply:
Ah! Of course. I've added a comment.

Reply:
``battle.undoChoice()`` could be changed to error if ``side.actions.length === 0``.

FWIW, in normal play, that code path would be taken if a player sends a bad decision and tries to undo it immediately afterwards -before the room battle is aware that the decision was faulty.